### ISSUE TITLE
Exception on deserializing empty element with an xsi attribute.

### ISSUE THREAD

**Issue by chikim79** [FasterXML/jackson-dataformat-xml#167]
Jackson XML fails with following exception when trying to deserialize an empty element with xsi attribute.

```
<a xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
       <b/>  <!-- works fine -->
       <c xsi:nil="false">0920</c> <!-- works fine -->
       <d xsi:nil="true"/> <!-- throws exception -->
</a>
```

```
public class A {
    String b;
    String c;
    String d;
    ...
}
```

```
com.fasterxml.jackson.databind.JsonMappingException: Can not deserialize instance of java.lang.String out of START_OBJECT token
 at [Source: java.io.StringReader@7862f584; line: 4, column: 8] (through reference chain: A["d"])
	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:148)
	at com.fasterxml.jackson.databind.DeserializationContext.mappingException(DeserializationContext.java:854)
	at com.fasterxml.jackson.databind.deser.std.StringDeserializer.deserialize(StringDeserializer.java:62)
	at com.fasterxml.jackson.databind.deser.std.StringDeserializer.deserialize(StringDeserializer.java:11)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:520)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:95)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:256)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:125)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3731)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2724)
```

I tried looking at the code, but couldn't see a straight forward way to fix the issue.

**Comment by Typografikon** [comment]
I have the same problem, any idea to deal with it?

**Comment by chikim79** [comment]
I regretably ended up using a different xml parser. Still I would love for Jackson to fix this.

**Comment by cowtowncoder** [comment]
Looking into this now. Can't fix in 2.6, but might be able to fix in 2.7.

**Comment by cowtowncoder** [comment]
And yes, figured out a way to get it fixed.
